### PR TITLE
Decouple Identity provider mappers from RealmModel

### LINKED ISSUE #31731: [Scalability of IDP Mappers] Add IDP Mapper methods to the IDPProvider
### Description

Methods similar to the the ones that exist in `RealmModel` to manage IDP Mappers must be added to the new IDPProvider, allowing the creation, update, removal and search of IDP Mappers in that provider instead of the realm.

### LINKED ISSUE #31732: [Scalability of IDP Mappers] Add implementation for IDP Mappers in JPA IDPProvider
### Description

This task is about adding the JPA implementation for the new IDP Mapper methods in JPA IDPProvider, moving the code currently in the JAP `RealmAdapter`.

### LINKED ISSUE #31733: [Scalability of IDP Mappers] Add implementation for IDP Mappers in Infinispan IDPProvider
### Description

This task is about adding the infinispan implementation for the new IDP Mapper methods in the Infinispan IDPProvider, moving the code currently in the Infinispan RealmAdapter.

As part of this task, we must ensure the mappers are no longer present in the `CachedRealm`. Also, the realm should no longer be invalidated when mappers are created/modified/deleted.

### LINKED ISSUE #31734: [Scalability of IDP Mappers] Deprecate IDP Mapper methods in RealmModel
### Description

Once the IDP Mapper are manageable through the IDPProvider, we must deprecate the methods in `RealmModel` and change their implementations so they delegate the operations to the new provider.

### LINKED ISSUE #31735: [Scalability of IDP Mappers] Replace calls to the IDP Mapper methods in RealmModel
### Description

All internal calls to the IDP Mapper methods in `RealmModel` should be replaced by calls to the new provider instead. When doing this extra care must be taken with methods that fetch all IDP mappers and then filter them in code. In these cases, smarter variant of the search methods available in the new provider must be used.

### PR DESCRIPTION
**Pull request by vramik**
Closes https://github.com/keycloak/keycloak/issues/31731
Closes https://github.com/keycloak/keycloak/issues/31732
Closes https://github.com/keycloak/keycloak/issues/31734
Closes https://github.com/keycloak/keycloak/issues/31733
Closes https://github.com/keycloak/keycloak/issues/31735

### CODE REVIEW

**Review by sguilhen — APPROVED** [review]
Great work, @vramik ! LGTM

File: model/jpa/src/main/java/org/keycloak/models/jpa/JpaIDPProvider.java [hunk 1]
```diff
@@ -324,6 +330,104 @@ public long count() {
...
+    public Stream<IdentityProviderMapperModel> getMappersStream() {
+        CriteriaBuilder builder = em.getCriteriaBuilder();
+        CriteriaQuery<IdentityProviderMapperEntity> query = builder.createQuery(IdentityProviderMapperEntity.class);
+        Root<IdentityProviderMapperEntity> mapper = query.from(IdentityProviderMapperEntity.class);
+
+        Predicate predicate = builder.equal(mapper.get("realmId"), getRealm().getId());
+
+        TypedQuery<IdentityProviderMapperEntity> typedQuery = em.createQuery(query.select(mapper).where(predicate));
+
+        return typedQuery.getResultStream().map(this::toModel);
```
Comment (pedroigor) [inline model/jpa/src/main/java/org/keycloak/models/jpa/JpaIDPProvider.java:413]: Don't we need to wrap the stream using `closing`? Same for other query methods.

File: server-spi-private/src/main/java/org/keycloak/broker/provider/mappersync/ConfigSyncEventListener.java [hunk 2]
```diff
@@ -51,12 +51,7 @@ public void onEvent(ProviderEvent event) {
                 LOG.debugf("Synchronizer %s matches event: %s", configSynchronizer, event);
 
                 if (realmMappers == null) {
-                    /*
-                     * an event always refers to just one realm, so we can use an arbitrary synchronizer to extract the
-                     * realm
-                     */
-                    RealmModel realm = configSynchronizer.extractRealm(event);
-                    realmMappers = realm.getIdentityProviderMappersStream()
+                    realmMappers = configSynchronizer.getKeycloakSession(event).identityProviders().getMappersStream()
```
Comment (pedroigor) [inline server-spi-private/src/main/java/org/keycloak/broker/provider/mappersync/ConfigSyncEventListener.java:54]: Do we have a follow-up issue to sort out this usage of `getMappersStream`?

Comment (sguilhen) [reply server-spi-private/src/main/java/org/keycloak/broker/provider/mappersync/ConfigSyncEventListener.java:54]: This one? https://github.com/keycloak/keycloak/issues/32233

File: model/jpa/src/main/java/org/keycloak/models/jpa/JpaIDPProvider.java (hunk 1 above)
Comment (sguilhen) [reply model/jpa/src/main/java/org/keycloak/models/jpa/JpaIDPProvider.java:413]: Good point, we probably need it, yes

Comment (vramik) [reply model/jpa/src/main/java/org/keycloak/models/jpa/JpaIDPProvider.java:413]: Will add it, thanks

**Review by sguilhen — APPROVED** [review]

**Comment by sguilhen** [comment]
@pedroigor Can you re-run the failing tests? Failures don't look related to the PR

**Review by pedroigor — APPROVED** [review]

**Pull request merged**